### PR TITLE
[MONDRIAN-2633] - Regression: Aggregate Tables not hit when Using a D…

### PR DESCRIPTION
**Pull request by cravobranco**
…imension Filter

* Fixed a side effect introduced in MONDRIAN-2285.
* Improved mechanism in aggregate table: a constraint will not be generated because name column is not the same as key column.

### CODE REVIEW

**Comment by cravobranco** [comment]
@lucboudreau , @kurtwalker , @mkambol , @pentaho-lmartins please review and merge if it's all ok. Thanks!

File: mondrian/src/main/java/mondrian/rolap/SqlConstraintUtils.java [hunk 1]
```diff
@@ -1674,6 +1674,12 @@ public static String constrainLevel(
                 // as the key column
                 int bitPos = column.getBitPosition();
                 AggStar.Table.Column aggColumn = aggStar.lookupColumn(bitPos);
+
+                if(aggColumn == null){
+                    LOG.warn("Aggregate table: A constraint will not be generated because name column is not the same as key column");
```
Comment (lucboudreau) [inline mondrian/src/main/java/mondrian/rolap/SqlConstraintUtils.java]: This message should be externalized in ``mondrianresource.xml``

Comment (cravobranco) [reply mondrian/src/main/java/mondrian/rolap/SqlConstraintUtils.java]: Agree, thanks!

**Comment by cravobranco** [comment]
@lucboudreau , thanks for your comments. PR is now updated so it's available again for review.

**Pull request merged**